server/src/main/java/org/opensearch/identity/ExtensionTokenProcessor.java

```diff
@@ -14,26 +37,80 @@
 public class ExtensionTokenProcessor {
     public static final String INVALID_TOKEN_MESSAGE = "Token must not be null and must be a colon-separated String";
     public static final String INVALID_EXTENSION_MESSAGE = "Token passed here is for a different extension";
+    public static final String INVALID_ALGO_MESSAGE = "Failed to create a token because an invalid hashing algorithm was used.";
+    public static final String INVALID_PRINCIPAL_MESSAGE = "Token passed here is for a different principal.";
+
+    public static final String ALGORITHM = "AES";
+    public static final int KEY_SIZE = 128;
+    public static final int INITIALIZATION_VECTOR_SIZE = 96;
+    public static final int TAG_LENGTH = 128;
+    
+    private final byte[] extensionUniqueId;
```

Comment:
The `extensionUniqueId` should remain a String.

Reply:
It has to be a byte[] to encode it but I switched the name of the attribute to avoid the confusion with the argument being passed in (which will be a string as you mentioned).